task-driver/src/settle_match_internal.rs

```diff
@@ -336,193 +266,53 @@ impl SettleMatchInternalTask {
-
-        let party0_commitment_proof = &self.order1_proof.commitment_proof;
-        let party1_commitment_proof = &self.order2_proof.commitment_proof;
-
-        apply_match_to_wallets(
-            &mut party0_modified_shares,
-            &mut party1_modified_shares,
-            party0_commitment_proof,
-            party1_commitment_proof,
-            &self.match_result,
-        );
-
-        let statement = ValidSettleStatement {
-            party0_modified_shares,
-            party1_modified_shares,
-            party0_send_balance_index: party0_commitment_proof.statement.balance_send_index,
-            party0_receive_balance_index: party0_commitment_proof.statement.balance_receive_index,
-            party0_order_index: party0_commitment_proof.statement.order_index,
-            party1_send_balance_index: party1_commitment_proof.statement.balance_send_index,
-            party1_receive_balance_index: party1_commitment_proof.statement.balance_receive_index,
-            party1_order_index: party1_commitment_proof.statement.order_index,
-        };
-
-        // Enqueue a job with the proof generation module
-        let (response_sender, response_receiver) = oneshot::channel();
-        self.proof_manager_work_queue
-            .send(ProofManagerJob {
-                type_: ProofJob::ValidSettle { witness, statement },
-                response_channel: response_sender,
-            })
-            .map_err(|_| {
-                SettleMatchInternalTaskError::EnqueuingJob(ERR_ENQUEUING_JOB.to_string())
-            })?;
-
-        // Await a response
-        let proof = response_receiver.await.map_err(|_| {
-            SettleMatchInternalTaskError::EnqueuingJob(ERR_AWAITING_PROOF.to_string())
-        })?;
-        self.valid_settle = Some(proof.into());
 
+        self.proof_bundle = Some(proof.into());
         Ok(())
     }
 
     /// Submit the match transaction
     async fn submit_match(&mut self) -> Result<(), SettleMatchInternalTaskError> {
         // Submit a `match` transaction
-        let party0_reblind_proof = &self.order1_proof.reblind_proof.statement;
-        let party1_reblind_proof = &self.order2_proof.reblind_proof.statement;
-        let valid_match_proof = self.valid_match_mpc.clone().unwrap();
-        let valid_settle_proof = self.valid_settle.clone().unwrap();
-
-        let tx_submit_res = self
-            .starknet_client
-            .submit_match(
-                party0_reblind_proof.original_shares_nullifier,
-                party1_reblind_proof.original_shares_nullifier,
-                party0_reblind_proof.reblinded_private_share_commitment,
-                party1_reblind_proof.reblinded_private_share_commitment,
-                valid_settle_proof.statement.party0_modified_shares.clone(),
-                valid_settle_proof.statement.party1_modified_shares.clone(),
+        let match_settle_proof = self.proof_bundle.clone().unwrap();
```

Comment:
can we `take` here to avoid a `clone`, or will we need the proof bundle in subsequent steps?